python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -138,32 +261,31 @@ def __init__(self, *args, **kwargs):
 
     @pipeBase.timeMethod
     def run(self, dataRef, selectDataList=[]):
-        """Assemble a coadd from a set of coaddTempExp
-
-        The coadd is computed as a mean with optional outlier rejection.
-
-        AssembleCoaddTask performs coaddition of "coadd temporary exposures" ("coaddTempExps").  Each
-        coaddTempExp is the size of a patch and contains data for one run, visit or (for a non-mosaic camera)
-        Exposure. The coaddTempExps to coadd are selected from the provided selectDataList based on their
-        overlap with the patch specified by dataRef.
-
-        By default, coaddTempExps contain backgrounds and hence require config.doMatchBackgrounds=True.
-        Background-subtracted coaddTempExps can be coadded by setting config.doMatchBackgrounds=False.
-
-        When background matching is enabled, the task may be configured to automatically select a reference
-        exposure (config.autoReference=True). If this is not done, then we require that the input dataRef
-        provides access to a coaddTempExp (dataset type coaddName + 'Coadd_tempExp') which is used as the
-        reference exposure.
-
-        @param dataRef: Data reference defining the patch for coaddition and the reference coaddTempExp
+        """!
+        \brief Assemble a coadd from a set of coaddTempExp
+
+        Get the skymap, information about the tract & patch to be coadded, the wcs & the outer 
+        bbox of the patch specified by the dataRef using \ref getSkyInfo. Pick the exposures to coadd for the 
+        specified patch using \ref selectExposures & make the list of corresponding coaddTempExps using 
+        \ref getTempExpRefList. Compute weights and inputScalers for each coaddTempExp using 
+        \ref prepareInputs. If config.doMatchBackground=True, get a background imageScaler object using 
+        \ref getBackgroundReferenceScaler and match backgrounds to the reference using 
+        \ref backgroundMatching. Assemble the coaddTempExps using \ref assemble. If background matching was 
+        performed, write the appropriate background matching metadata using 
+        \ref addBackgroundMatchingMetadata. If config.doInterp=True, interpolate over NaNs using the
+        \ref InterpImageTask_ "interpImage" subtask. If config.doWrite=True, call \ref writeCoaddOutput to 
+        write out the coadd. Return the coadd.
```

Comment:
I feel like there is too much detail in this description.  I think it could be condensed to only include the important features instead of going through the code line by line.